distribution/docs/src/main/resources/_contents/_endpoints/cometd-endpoint-contents.adoc

```diff
@@ -0,0 +1,845 @@
+
+==== CometD Endpoint
+
+The http://cometd.org[CometD] endpoint enables asychronous search capabilties.
+The CometD protocol is used to execute searches, retrieve results, and receive notifications.
+
+For an example of using CometD within a webapp see: `distribution/sdk/sample-cometd/`
+
+===== Installing CometD Endpoint
+
+The CometD Endpoint is installed by default with a standard installation in the ${ddf-ui} application.
+
+===== Configuring CometD Endpoint
+
+None.
+
+===== Using CometD Endpoint
+
+.CometD Endpoint URL
+----
+${secure_url}/search/cometd
+----
+
+====== Query
+
+Queries can be executed over CometD using the `/service/query` channel.
+Query messages are json formatted and use cql alongside several other parameters.
+
+.Query Parameters
+[cols="1,3,1", options="header"]
+|===
+|Parameter Name
+|Description
+|Required
+
+|`src`
+|Catalog Source
+|No
+
+|`cql`
+|CQL query
+|Yes
+
+|`sort`
+|Sort Type
+|No
+
+|`id`
+|Query ID (Should be a uuid), This determines the channel that the query results will be returned on.
+|Yes
+|===
+
+Before a query is published the client should subscribe to the channel that will be passed in to the `id` field in order to receive query results once the query is executed.
+
+For example if the following id was generated `3b19bc9c-2155-4ca6-bae8-65a9c8e373f6`, the client should subscribe to `/3b19bc9c-2155-4ca6-bae8-65a9c8e373f6`
+
+Then the following example query could be executed:
+
+.`/service/query`
+[source,json]
+----
+{
```

Comment:
The example should include all possible options, i.e., `src`, `cql`, `sort` and `id`.